packages/components/src/components/arrow-indicator/arrow-indicator.tsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import Icon from '../icon/icon';
+
+type TArrowIndicatorProps = {
+    classname?: string;
+    value?: number | string;
+};
+
+type TData = {
+    previous_icon: string;
+    icon: string;
+    previous_value?: string | number;
+    value?: string | number;
+};
+
+const ArrowIndicator = ({ classname, value }: TArrowIndicatorProps) => {
+    const [is_visible, setIsVisible] = React.useState(false);
+    const [data, setData] = React.useState<TData>({
+        icon: '',
+        previous_icon: '',
+    });
+    const { icon, previous_icon, previous_value } = data;
+    const has_comparable_values = !isNaN(Number(data.value)) && !isNaN(Number(value));
+    const timeout_id = React.useRef<ReturnType<typeof setTimeout>>();
+    React.useEffect(() => {
+        setIsVisible(true);
```

Comment:
should we move this line inside the if condition below to make sure we only show it when there is an actual change in the value?

Reply:
@balakrishna-deriv the value that comes is always different from the previous one since the component is memoized. will we still need the if condition in this case? :)